CMakeLists.txt

```diff
@@ -112,9 +112,59 @@ message(STATUS "Project name: ${PROJECT_NAME}")
 message(STATUS "Project version: ${PROJECT_VERSION_STRING_FULL}")
 
 if(COMPILE_WASM)
+  # See https://github.com/emscripten-core/emscripten/blob/main/src/settings.js
   set(WORMHOLE ON CACHE BOOL "Use WASM wormhole in intgemm https://bugzilla.mozilla.org/show_bug.cgi?id=1672160")
-  list(APPEND WASM_COMPILE_FLAGS -O3 -g2 -fPIC -mssse3 -msimd128)
-  list(APPEND WASM_COMPILE_FLAGS "SHELL:-s WASM=1" "SHELL:-s ASSERTIONS=0" "SHELL:-s DISABLE_EXCEPTION_CATCHING=1" "SHELL:-s LLD_REPORT_UNDEFINED" "SHELL:-s FORCE_FILESYSTEM=1" "SHELL:-s ALLOW_MEMORY_GROWTH=1")
+  list(APPEND WASM_COMPILE_FLAGS
+    -O3
+    -g2
+    -fPIC
+    -mssse3
+    -msimd128
+    # -fno-exceptions # Can't do that because spdlog uses exceptions
+    -sDISABLE_EXCEPTION_CATCHING=1
+    -sSTRICT=1
+  )
+  list(APPEND WASM_LINK_FLAGS
+    --bind # TODO: I thought it was an alias for -lembind, but maybe not yet in this version of emscripten
+    -g2
+    
```

Comment:
The optimization flag is a compile + link flag. So, we should add `-O3` in link flag as well.